From b716051fec9c258efe9cb9bc00e831d28dfc422f Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa@mageia.org>
Date: Sun, 28 May 2017 13:56:45 -0400
Subject: [PATCH 1002/1002] Run the makecache service+timer only if the DNF
 cache exists

We only want the makecache service and timer to activate if the
user has invoked DNF once and caused the cache to be created.
Otherwise, this should just not get invoked.
---
 etc/systemd/dnf-makecache.service | 3 +++
 etc/systemd/dnf-makecache.timer   | 2 ++
 2 files changed, 5 insertions(+)

diff --git a/etc/systemd/dnf-makecache.service b/etc/systemd/dnf-makecache.service
index afc3bc5..98e4438 100644
--- a/etc/systemd/dnf-makecache.service
+++ b/etc/systemd/dnf-makecache.service
@@ -4,6 +4,9 @@ Description=dnf makecache
 # while someone might theoretically want the cache updated, in practice
 # anyone who wants that could override this via a file in /etc.
 ConditionPathExists=!/run/ostree-booted
+# If DNF has never been used, we don't really want to run the service.
+# Once it has been run once and the cache has been fully generated, then we do.
+ConditionPathExists=/var/cache/dnf/packages.db
 
 After=network-online.target
 
diff --git a/etc/systemd/dnf-makecache.timer b/etc/systemd/dnf-makecache.timer
index 7c0838b..ca071bc 100644
--- a/etc/systemd/dnf-makecache.timer
+++ b/etc/systemd/dnf-makecache.timer
@@ -5,6 +5,8 @@ ConditionKernelCommandLine=!rd.live.image
 ConditionPathExists=!/run/ostree-booted
 # Disable timer for live environments
 ConditionPathExists=!/run/mgalive
+# See comment in dnf-makecache.service
+ConditionPathExists=/var/cache/dnf/packages.db
 Wants=network-online.target
 
 [Timer]
-- 
2.13.6

